File: java/111.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int min = Integer.MAX_VALUE;
    public int minDepth(TreeNode root) {
        if(root == null){
            return 0;
        }
        depth(root,0);
        return min;
    }
    public void depth(TreeNode root,int de){
        if(root == null){
            return;
        }
        if(root.left == null && root.right == null){
            min = Math.min(min,de+1);
            return;
        }
        depth(root.left,de+1);
        depth(root.right,de+1);
    }
}

File: go/111.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func minDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}

	return helper(root)
}
func helper(root *TreeNode) int {
	if root == nil {
		return int(^uint(0) >> 1)
	}
	if root.Left == nil && root.Right == nil {
		return 1
	}
	left := helper(root.Left)
	right := helper(root.Right)

	return 1 + min(left, right)
}
func min(i, j int) int {
	if i < j {
		return i
	}
	return j
}
